vm/src/stdlib/builtins.rs

```diff
@@ -146,20 +161,23 @@ mod builtins {
             }
             #[cfg(feature = "rustpython-parser")]
             {
-                use crate::{builtins::PyBytesRef, convert::ToPyException};
+                use crate::convert::ToPyException;
                 use num_traits::Zero;
                 use rustpython_parser as parser;
 
-                let source = Either::<PyStrRef, PyBytesRef>::try_from_object(vm, args.source)?;
+                let source = ArgStrOrBytesLike::try_from_object(vm, args.source)?;
+                let source = source.borrow_bytes();
+
                 // TODO: compiler::compile should probably get bytes
-                let source = match &source {
-                    Either::A(string) => string.as_str(),
-                    Either::B(bytes) => std::str::from_utf8(bytes)
-                        .map_err(|e| vm.new_unicode_decode_error(e.to_string()))?,
-                };
+                let source = std::str::from_utf8(&source)
+                    .map_err(|e| vm.new_unicode_decode_error(e.to_string()))?;
 
                 let flags = args.flags.map_or(Ok(0), |v| v.try_to_primitive(vm))?;
 
+                if !(flags & !ast::PY_COMPILE_FLAGS_MASK).is_zero() {
+                    return Err(vm.new_value_error("compile() unrecognized flags".to_owned()));
```

Comment:
This line seems to be related to `test_future.test_attributes`. Probably compile accepts more flags than AST.

Reply:
it looks like there are some undocumented flags that the stdlib depends on. I've added them, but not made them `pub` so people don't use them.